provider/oci/init.go

```diff
@@ -0,0 +1,14 @@
+// Copyright 2017 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package oci
+
+import "github.com/juju/juju/environs"
+
+const (
+	providerType = "oci"
+)
+
+func init() {
+	environs.RegisterProvider(providerType, &EnvironProvider{})
```

Comment:
Do we want to have the oci provider behind a feature flag while we land bits?

Reply:
Given this is "oci" and not "oracle", I think our general policy is that we can land code for new providers as long as it doesn't change the behavior elsewhere. Only people opting into the new provider would actually be affected.

Arguably we want to signal that a given provider isn't ready for production use (we won't support upgrading a controller that is using the old code).
Could we do that as just a warning during bootstrap/add-model? or is a feature-flag a better way to convey that information?